src/utils/loadAndCompressIfNecessary.js

```diff
@@ -1,37 +1,25 @@
 import { gzip } from 'fflate';
-import { Buffer } from 'buffer/';
-
-function uintArrayToBuffer(array) {
-  return array.buffer.slice(array.byteOffset, array.byteLength + array.byteOffset);
-}
-
-const loadAndCompressIfNecessary = async (bundle, onCompression = () => { }) => {
-  const inGzipFormat = ['application/gzip', 'application/x-gzip'].includes(bundle.type);
-
-  return new Promise((resolve, reject) => {
-    const reader = new FileReader();
-    reader.onabort = () => reject(new Error('aborted'));
-    reader.onerror = () => reject(new Error('error'));
-    reader.onload = () => {
-      const loadedFile = reader.result;
-
-      if (inGzipFormat) {
-        resolve(loadedFile);
-      } else {
-        onCompression();
-
-        const loadedFileUint = Buffer.from(loadedFile);
-
-        gzip(loadedFileUint, {}, (error, compressedFile) => {
-          if (error) { reject(new Error('error')); }
-
-          resolve(uintArrayToBuffer(compressedFile));
-        });
-      }
-    };
-
-    reader.readAsArrayBuffer(bundle);
+import readFileToBuffer from './readFileToBuffer';
+
+const compress = (buffer) => new Promise((resolve, reject) => {
```

Comment:
I see there are a bunch of helper methods in utils for data upload, like `src/utils/readFileToBuffer.js` and this one for example. Why don't you create a new folder in `utils` for `data_upload` and move these files there, so that they are easier to find?